metabase/metabase
# Use `DataPicker` in `JoinTablePicker`

Closes #40952

### How to verify

1. Create a new question and choose data source
2. Try to join something to it
    - the selection of questions/tables/models should be limited to the database of data source selected in step 1
    - the database selection step in "Tables" tab should not be visible
    - "Models" and "Saved questions" tabs should not be visible if data source database does not have any models or questions respectively

Note: changing data source of a question to a one from a different database should remove the no-longer-valid join clause, but it does not work - I reported it separately: #42385

### Demo

https://github.com/metabase/metabase/assets/6830683/00a1098d-2815-42eb-a82e-352505828227
